File: warranty-service/src/domain/CoverageStatus.ts
/**
 * Value Object: CoverageStatus
 */
export const CoverageStatus = {
  PENDING: 'PENDING',
  COVERED: 'COVERED',
  REJECTED: 'REJECTED',
} as const;

export type CoverageStatus = (typeof CoverageStatus)[keyof typeof CoverageStatus];

File: _java-archive/warranty-service/src/main/java/dev/kritchalach/warranty/domain/CoverageStatus.java
package dev.kritchalach.warranty.domain;

/**
 * Value Object: CoverageStatus
 * ผลการตรวจสอบความคุ้มครอง warranty
 */
public enum CoverageStatus {
    PENDING,    // เพิ่งสร้าง claim ยังไม่ได้ตรวจ
    COVERED,    // อยู่ใน scope + ยังไม่หมดอายุ → คุ้มครอง
    REJECTED    // หมดอายุ หรือ defect type ไม่อยู่ใน scope
}
